build/configs/rtl8721csm/tc/defconfig

```diff
@@ -638,7 +638,7 @@ CONFIG_NET_MEM_ALIGNMENT=4
 # CONFIG_NET_MEM_LIBC_MALLOC is not set
 CONFIG_NET_MEMP_MEM_MALLOC=y
 # CONFIG_NET_MEM_USE_POOLS is not set
-CONFIG_NET_MEM_SIZE=153600
+CONFIG_NET_MEM_SIZE=81920
```

Comment:
Doesn't it degrade wifi performance? how about increase internal ram size for internal usages?
We can use entire SRAM for internal usages, not for heap

Reply:
wifi performance related parameters are tuned only on hello config and can retain the performance with this change(already tuned on #4668 ). So just make the code fit into sram we are modifying this parameter on other examples.
we can still use the entire SRAM, because now only the left space will be used for heap. we didn't fix any size for heap and it is determined by the linker.

Reply:
You modify this change on all of defconfig and this is the buffer of network. It could affect network performance. Could you let use know wifi performance test result?

Reply:
Wifi performance is still the same on hello config with TX: ~22 Mbps and RX: ~22 Mbps, as there is no modification on hello config part in this PR.

Reply:
I don't know why you mentioned hello config for wifi performance measurement. We don't use hello config for that. Anyone we use should have the same wifi performance. We will not use hello config for the project, loadable_apps is using.

Reply:
I see, then I'll check performance measurement on loadable apps and let you know.

Reply:
As I said first, we don't need to use remain SRAM for heap so that we can increase reserved SRAM size for this issue. If we can apply the same code for all of defconfig and it will resolve this issue, we can increase it.

Reply:
@sunghan-chang, we didn't reserve any SRAM for heap. We got the build error because enabling SE together with default CONFIG_NET_MEM_SIZE exceeds the maximum size of SRAM. So to resolve this we only need to decrease the size of  NET_MEM_SIZE which we already tested and will not hinder the target wifi performance.

Reply:
os_heap_init( ) which is in rtl8721dhp_app_start.c does.
Anyway, could you confirm below?
1. Sum of Below is 512K. Realtek uses entire SRAM for Realtek usages. Is this right?
```
	ROMBSS_RAM_COM (rw)  : 		ORIGIN = 0x10000000, LENGTH = 0x10001000 - 0x10000000	/* ROM BSS COMMON(S & NS both used) RAM: 4K */
	ROMBSS_RAM_NS (rw)  : 		ORIGIN = 0x10001000, LENGTH = 0x10002000 - 0x10001000	/* ROM BSS NS RAM: 4K */
	RSVD_RAM_NS (rw)  : 		ORIGIN = 0x10002000, LENGTH = 0x10004000 - 0x10002000	/* RSVD RAM: 8K */
	MSP_RAM_NS (rw)  : 		ORIGIN = 0x10004000, LENGTH = 0x10005000 - 0x10004000	/* MSP_NS RAM: 4K */

	BD_RAM_NS (rwx)  : 		ORIGIN = 0x10005000, LENGTH = 0x10077000 - 0x10005000	/* MAIN RAM NS: 456K */
	BD_RAM_S (rwx)  : 		ORIGIN = 0x10077000, LENGTH = 0x1007B000 - 0x10077000	/* MAIN RAM S: 16K */
	BD_RAM_NSC (rwx)  : 		ORIGIN = 0x1007B000, LENGTH = 0x1007C000 - 0x1007B000	/* MAIN RAM NSC: 4K */
	ROMBSS_RAM_S (rwx)  : 		ORIGIN = 0x1007C000, LENGTH = 0x1007D000 - 0x1007C000	/* ROM BSS RAM S: 4K */
	BOOTLOADER_RAM_S (rwx)  : 	ORIGIN = 0x1007D000, LENGTH = 0x1007F000 - 0x1007D000	/* BOOT Loader RAM: 8K */
	MSP_RAM_S (rwx)  : 		ORIGIN = 0x1007F000, LENGTH = 0x10080000 - 0x1007F000	/* MSP_S RAM: 4k */
```
2. As far as I know, that config is the buffer of network stack and it affects throughput. Please let me know test wifi performance result with changed value (81920) and unchanged value (153600) in the same base.

Reply:
> os_heap_init( ) which is in rtl8721dhp_app_start.c does.
> Anyway, could you confirm below?
> 
> 1. Sum of Below is 512K. Realtek uses entire SRAM for Realtek usages. Is this right?
> 
> ```
> 	ROMBSS_RAM_COM (rw)  : 		ORIGIN = 0x10000000, LENGTH = 0x10001000 - 0x10000000	/* ROM BSS COMMON(S & NS both used) RAM: 4K */
> 	ROMBSS_RAM_NS (rw)  : 		ORIGIN = 0x10001000, LENGTH = 0x10002000 - 0x10001000	/* ROM BSS NS RAM: 4K */
> 	RSVD_RAM_NS (rw)  : 		ORIGIN = 0x10002000, LENGTH = 0x10004000 - 0x10002000	/* RSVD RAM: 8K */
> 	MSP_RAM_NS (rw)  : 		ORIGIN = 0x10004000, LENGTH = 0x10005000 - 0x10004000	/* MSP_NS RAM: 4K */
> 
> 	BD_RAM_NS (rwx)  : 		ORIGIN = 0x10005000, LENGTH = 0x10077000 - 0x10005000	/* MAIN RAM NS: 456K */
> 	BD_RAM_S (rwx)  : 		ORIGIN = 0x10077000, LENGTH = 0x1007B000 - 0x10077000	/* MAIN RAM S: 16K */
> 	BD_RAM_NSC (rwx)  : 		ORIGIN = 0x1007B000, LENGTH = 0x1007C000 - 0x1007B000	/* MAIN RAM NSC: 4K */
> 	ROMBSS_RAM_S (rwx)  : 		ORIGIN = 0x1007C000, LENGTH = 0x1007D000 - 0x1007C000	/* ROM BSS RAM S: 4K */
> 	BOOTLOADER_RAM_S (rwx)  : 	ORIGIN = 0x1007D000, LENGTH = 0x1007F000 - 0x1007D000	/* BOOT Loader RAM: 8K */
> 	MSP_RAM_S (rwx)  : 		ORIGIN = 0x1007F000, LENGTH = 0x10080000 - 0x1007F000	/* MSP_S RAM: 4k */
> ```

No, Realtek doesn't uses the entire SRAM. data/bss of TizenRT are also under SRAM as below,

![sram_data_bss](https://user-images.githubusercontent.com/31299557/102990452-5a53ac80-4552-11eb-892e-1dcefc5e56fb.PNG)

Reply:
> 2. As far as I know, that config is the buffer of network stack and it affects throughput. Please let me know test wifi performance result with changed value (81920) and unchanged value (153600) in the same base.

Please find the result based on base <a42efcf4ef25d2763cd884e74796778ce741f57c> as below,

CONFIG_NET_MEM_SIZE(hello config) | 81900 | 153600
-- | -- | --
TX (Mbps) | ~22 | ~14
RX (Mbps) | ~22 | ~10

current hello config with CONFIG_NET_MEM_SIZE=81920 is already the best balance version we have got from previous tuning experiments.

Reply:
Please let me confirm test conditional. Did you use exactly the same configs for others except CONFIG_NET_MEM_SIZE?

Reply:
Yes, the test conditional are all same except for CONFIG_NET_MEM_SIZE.